subsys/bluetooth/audio/cap_commander.c

```diff
@@ -399,14 +400,302 @@ int bt_cap_commander_broadcast_reception_start(
 
 	return 0;
 }
-#endif /* CONFIG_BT_BAP_BROADCAST_ASSISTANT */
+
+static void
+copy_broadcast_reception_stop_param(struct bt_bap_broadcast_assistant_mod_src_param *mod_src_param,
+				    struct cap_broadcast_reception_stop *stop_param)
+{
+	mod_src_param->src_id = stop_param->src_id;
+	mod_src_param->pa_sync = false;
+	mod_src_param->pa_interval = BT_BAP_PA_INTERVAL_UNKNOWN;
+	mod_src_param->num_subgroups = stop_param->num_subgroups;
+
+	mod_src_param->subgroups = stop_param->subgroups;
+}
+
+static void cap_commander_broadcast_assistant_recv_state_cb(
+	struct bt_conn *conn, int err, const struct bt_bap_scan_delegator_recv_state *state)
+{
+	struct bt_cap_common_proc *active_proc = bt_cap_common_get_active_proc();
+	struct bt_conn *cap_conn = NULL;
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	uint16_t total_bis_sync;
+	uint8_t src_id = 0;
```

Comment:
Looks like several of these can be moved inside the `if (bt_cap_common_conn_in_active_proc(conn)` clause. Alternatively, you can do as we do in the CAP initiator with
```c
	if (!bt_cap_common_conn_in_active_proc(conn)) {

		/* State change happened outside of a procedure; ignore */
		return;
	}
```